### PR TITLE
fix: honor pageRanges when printing

### PR DESCRIPTION
**Pull request by codebytere**
Backport of https://github.com/electron/electron/pull/25064.

See that PR for more details.

Notes: Fixed an issue where pageRanges was not properly honored when printing.

### CODE REVIEW

**Review by zcbenz — CHANGES_REQUESTED** [review]
There are typescript errors:

```
ERROR in /Users/distiller/project/src/electron/lib/browser/api/web-contents.ts
[tsl] ERROR in /Users/distiller/project/src/electron/lib/browser/api/web-contents.ts(280,27)
      TS2339: Property 'from' does not exist on type 'PageRanges[]'.

ERROR in /Users/distiller/project/src/electron/lib/browser/api/web-contents.ts
[tsl] ERROR in /Users/distiller/project/src/electron/lib/browser/api/web-contents.ts(285,27)
      TS2339: Property 'to' does not exist on type 'PageRanges[]'.

ERROR in /Users/distiller/project/src/electron/lib/browser/api/web-contents.ts
[tsl] ERROR in /Users/distiller/project/src/electron/lib/browser/api/web-contents.ts(292,24)
      TS2339: Property 'from' does not exist on type 'PageRanges[]'.

ERROR in /Users/distiller/project/src/electron/lib/browser/api/web-contents.ts
[tsl] ERROR in /Users/distiller/project/src/electron/lib/browser/api/web-contents.ts(293,22)
      TS2339: Property 'to' does not exist on type 'PageRanges[]'.
```

**Comment by ckerr** [comment]
@codebytere is this PR still live?

**Comment by codebytere** [comment]
@ckerr i need to figure out the TS stuff but yes!

File: docs/api/web-contents.md [hunk 1]
```diff
@@ -1296,9 +1296,9 @@ Returns [`PrinterInfo[]`](structures/printer-info.md)
   * `pagesPerSheet` Number (optional) - The number of pages to print per page sheet.
   * `collate` Boolean (optional) - Whether the web page should be collated.
   * `copies` Number (optional) - The number of copies of the web page to print.
-  * `pageRanges` Record<string, number> (optional) - The page range to print.
-    * `from` Number - the start page.
-    * `to` Number - the end page.
+  * `pageRanges` Object[] (optional) - The page range to print.
```
Comment (jkleinsc) [inline docs/api/web-contents.md:1299]:
```suggestion
  * `pageRanges` Object[] (optional) - The page range to print. On macOS, only one range is honored.
```

**Review by jkleinsc — CHANGES_REQUESTED** [review]
Looks like a doc change was missed - other than that LGTM

**Review by 46967357 — APPROVED** [review]

**Review by zcbenz — APPROVED** [review]

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**